Extension/src/LanguageServer/settings.ts

```diff
@@ -382,10 +382,15 @@ export class CppSettings extends Settings {
     public set defaultCompilerPath(value: string | undefined) {
         const defaultCompilerPathStr: string = "default.compilerPath";
         const compilerPathInfo: any = this.Section.inspect(defaultCompilerPathStr);
-        this.Section.update(defaultCompilerPathStr, value,
-            compilerPathInfo.workspaceFolderValue !== undefined ? vscode.ConfigurationTarget.WorkspaceFolder :
-                (compilerPathInfo.workspaceValue !== undefined ? vscode.ConfigurationTarget.Workspace :
-                    vscode.ConfigurationTarget.Global));
+        let target: vscode.ConfigurationTarget = vscode.ConfigurationTarget.Global;
+        if (this.resource !== undefined) {
```

Comment:
Combine `this.resource !== undefined || compilerPathInfo.workspaceFolderValue !== undefined`?

Reply:
sure